### ISSUE TITLE
Use go-ipa for Multiproof point deserialization

### ISSUE THREAD

**Issue by jsign** [gballet/go-verkle#392]
Currently, we [have some custom code](https://github.com/gballet/go-verkle/blob/master/proof_ipa.go#L243-L252) to deserialize a `Multiproof`. In go-ipa, [there's a specific API](https://github.com/crate-crypto/go-ipa/blob/master/multiproof.go#L241) to do this deserialization.

Both styles are a bit different since go-ipa expects an opaque stream of bytes and does the deserialization. Here, in go-verkle, we have the serialized proof already in parts, which probably explains why we have some custom code.

In any case, we must rely on go-ipa to do any deserialization to know this is done safely. If that means changing go-ipa APIs since we need to pass these "parts", I think that's fine.  But let's try to rely on go-ipa for any security issues since it has (and will continue to have) more tests (and audits) to be sure these APIs are safe.

**Comment by gballet** [comment]
Yeah this is the old API, it has some issues. We would have to revisit it before we move back to this system. I don't oppose it, but let's keep things as they are now for kaustinen. It's much easier to debug.
